test/peer_prop.ml

```diff
@@ -0,0 +1,51 @@
+open Generator.Test
+open QCheck2.Gen
+
+module SUT = Pollinate.Peer
+
+let add_peer =
+  QCheck2.Test.make ~count:1000
+    ~name:"add_peer on empty neighbors leads to size of 1"
+    (pair peer_gen peer_gen) (fun (peer, neighbor) ->
+      let _ = SUT.add_neighbor peer neighbor in
+      Base.Hashtbl.length peer.neighbors == 1)
+
+let add_peers =
+  QCheck2.Test.make ~count:1000
+    ~name:
+      "add_peers results in neighors having the same length that given \
```

Comment:
This should NOT be true if any of the peers in the generated list have the same address, right?

Reply:
That's an other property, working on a dedicated test for it. Thanks for pointing it out 👍

Reply:
New property test added!